### PR TITLE
Makes ThreadLocalServerAndClientSpanState public and polishes use

### PR DESCRIPTION
**Pull request by adriancole**
ServerAndClientSpanState is awkward to use without examples. This shows
use of the initial state value `ServerSpan.EMPTY`, and polishes Brave's
builder, which accidentally implied creating state twice.

This change is paid for by removing `SimpleServerAndClientSpanState`,
which was unusable dead code due to package protection.

This also makes ThreadLocalServerAndClientSpanState public so it can be
used directly in Spring.

See #90

### CODE REVIEW

**Comment by adriancole** [comment]
@pbetkier you're probably interested in this.

Note I think we will need follow-up change if we want to make it possible for finagle and brave to collaborate (because finagle would need to create the currently hidden ServerSpan factory method).

Regardless, this is a first step, and helps in cases where you don't need to create your own server spans.

**Comment by adriancole** [comment]
updated to make ThreadLocalServerAndClientSpanState public, which avoids copy/pasting this for use in spring.

File: brave-core/src/main/java/com/github/kristofa/brave/ThreadLocalServerAndClientSpanState.java [hunk 1]
```diff
@@ -50,7 +50,6 @@ public ThreadLocalServerAndClientSpanState(InetAddress ip, int port, String serv
      * @param serviceName Name of the local service being traced. Should be lowercase and not <code>null</code> or empty.
      */
     public ThreadLocalServerAndClientSpanState(int ip, int port, String serviceName) {
-        Util.checkNotNull(ip, "ip address must be specified.");
```
Comment (adriancole) [inline brave-core/src/main/java/com/github/kristofa/brave/ThreadLocalServerAndClientSpanState.java]: int can't be null :P

**Comment by adriancole** [comment]
ps this is the only thing I'm actually blocked on. I found a way around needing to change Random.

**Comment by kristofa** [comment]
:+1:

**Pull request merged**